docs/bigquery/snippets.py

```diff
@@ -89,476 +97,556 @@ def do_something_with(_):
-
-    # [START dataset_create]
-    dataset = client.dataset(DATASET_NAME)
-    dataset.create()              # API request
-    # [END dataset_create]
-
-    to_delete.append(dataset)
+    DATASET_ID = 'create_dataset_%d' % (_millis(),)
 
+    # [START create_dataset]
+    dataset_ref = client.dataset(DATASET_ID)
+    dataset = bigquery.Dataset(dataset_ref)
+    dataset.description = 'my dataset'
+    dataset = client.create_dataset(dataset)  # API request
+    # [END create_dataset]
 
-def test_dataset_exists(client, to_delete):
-    """Test existence of a dataset."""
-    DATASET_NAME = 'dataset_exists_%d' % (_millis(),)
-    dataset = client.dataset(DATASET_NAME)
     to_delete.append(dataset)
 
-    # [START dataset_exists]
-    assert not dataset.exists()   # API request
-    dataset.create()              # API request
-    assert dataset.exists()       # API request
-    # [END dataset_exists]
-
 
-def test_dataset_reload(client, to_delete):
+def test_get_dataset(client, to_delete):
     """Reload a dataset's metadata."""
-    DATASET_NAME = 'dataset_reload_%d' % (_millis(),)
-    dataset = client.dataset(DATASET_NAME)
+    DATASET_ID = 'get_dataset_%d' % (_millis(),)
+    dataset_ref = client.dataset(DATASET_ID)
+    dataset = bigquery.Dataset(dataset_ref)
     dataset.description = ORIGINAL_DESCRIPTION
-    dataset.create()
+    dataset = client.create_dataset(dataset)  # API request
     to_delete.append(dataset)
 
-    # [START dataset_reload]
+    # [START get_dataset]
     assert dataset.description == ORIGINAL_DESCRIPTION
     dataset.description = LOCALLY_CHANGED_DESCRIPTION
     assert dataset.description == LOCALLY_CHANGED_DESCRIPTION
-    dataset.reload()              # API request
+    dataset = client.get_dataset(dataset)  # API request
     assert dataset.description == ORIGINAL_DESCRIPTION
-    # [END dataset_reload]
+    # [END get_dataset]
 
 
-def test_dataset_patch(client, to_delete):
-    """Patch a dataset's metadata."""
-    DATASET_NAME = 'dataset_patch_%d' % (_millis(),)
-    dataset = client.dataset(DATASET_NAME)
+def test_update_dataset_simple(client, to_delete):
+    """Update a dataset's metadata."""
+    DATASET_ID = 'update_dataset_simple_%d' % (_millis(),)
+    dataset = bigquery.Dataset(client.dataset(DATASET_ID))
     dataset.description = ORIGINAL_DESCRIPTION
-    dataset.create()
+    client.create_dataset(dataset)
     to_delete.append(dataset)
 
-    # [START dataset_patch]
-    ONE_DAY_MS = 24 * 60 * 60 * 1000
+    # [START update_dataset_simple]
     assert dataset.description == ORIGINAL_DESCRIPTION
-    dataset.patch(
-        description=PATCHED_DESCRIPTION,
-        default_table_expiration_ms=ONE_DAY_MS
-    )      # API request
-    assert dataset.description == PATCHED_DESCRIPTION
-    assert dataset.default_table_expiration_ms == ONE_DAY_MS
-    # [END dataset_patch]
+    dataset.description = UPDATED_DESCRIPTION
+
+    client.update_dataset(dataset, ['description'])  # API request
```

Comment:
This line should be `dataset = client.update_dataset(...)` because we want the user to get the updated `etag` value. `update_dataset` does not (or at least should not) mutate the arguments it is given. (Plus the assert below will make more sense.)

Reply:
Fixed in e30c19a